src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingSubmissionResource.java

```diff
@@ -399,7 +396,12 @@ public ResponseEntity<ProgrammingSubmission> lockAndGetProgrammingSubmission(@Pa
         // prepare programming submission for response
         programmingSubmissionService.hideDetails(programmingSubmission, user);
         // remove automatic results before sending to client
-        programmingSubmission.setResults(programmingSubmission.getManualResults());
+        if (correctionRound >= programmingSubmission.getManualResults().size()) {
```

Comment:
You could think about using the getter once instead of multiple times

Reply:
I think since we are only calling this getter twice, we can leave it as is. But yes, if we need to call it often in the future, we should extract it to a variable :+1:

Reply:
It's not only about performance :)
I never expect a getter to return the same result in a program with multiple threads.

(Nevertheless, in most cases they will return the same value :) )